docs/source/guides/user/chapters/configuring.rst

```diff
@@ -174,59 +137,119 @@ of precedence for variables (from least precedence to most):
+and command-line options to configure its behavior. It is important to have a
+very well defined system type for the configuration file and argument options.
+
+Although config files options and command-line arguments are always considered
+``strings``, you should give a proper format representation so those values can
+be parsed into a proper type internally on Avocado.
+
+Currently Avocado supports the following data types for the configuration options:
+``string``, ``integer``, ``float``, ``bool`` and ``list``. Besides those
+primitives data types Avocado also supports custom data types that can be used
+by a particular plugin.
+
+Bellow, you will find information on how to set options based on those basic
+data types using both: configuration files and command-line arguments.
+
+Strings
+~~~~~~~
+
+Strings are the basic ones and the syntax is the same for both: configuration
+files and command-line arguments: Just the string that can be inside ``""`` or
+``''``.
+
+Example using the configuration file:
+
+.. code-block:: ini
+
+  [foo]
+  bar = 'hello world'
 
-When running tests, we are frequently looking to:
+Example using the command-line:
 
-* Locate tests
-* Write logs to a given location
-* Grab files that will be useful for tests, such as ISO files or VM disk
-  images
+.. code-block:: bash
 
-Avocado has a module dedicated to find those paths, to avoid cumbersome
-path manipulation magic that people had to do in previous test frameworks [#f1]_.
+  $ avocado run --foo bar /bin/true
 
-If you want to list all relevant directories for your test, you can use
-`avocado config --datadir` command to list those directories. Executing
-it will give you an output similar to the one seen below::
+Integers
+~~~~~~~~
+
+Integer numbers are as simple as strings.
+
+Example using the configuration file:
+
+.. code-block:: ini
+
+  [run]
+  job_timeout = 60
+
+Example using the command-line:
+
+.. code-block:: bash
+
+  $ avocado run --job-timeout 50 /bin/true
+
+Floats
+~~~~~~
+
+Float numbers has the same representation as integers, but you should use `.`
+(dot) to separate the decimals. i.e: `80.3`.
+
+Booleans
+~~~~~~~~
+
+When talking about configuration files, accepted values for a boolean option
+are '1', 'yes', 'true', and 'on', which cause this method to return True, and
+'0', 'no', 'false', and 'off', which cause it to return False. But, when
+talking about command-line, booleans options don't need any argument, the
+option itself will enable or disable the settings, depending on the context.
+
+Example using the configuration file:
+
+.. code-block:: ini
+
+  [core]
+  verbose = 'true'
+
+Example using the command-line:
+
+.. code-block:: bash
+
+  $ avocado run --verbose /bin/true
+
+.. note:: Currently we still have some "old style boolean" options where you
+  should pass "on" or "off" on the command-line. i.e: ``--sysinfo=off``. Those
+  options are going to be replaced soon.
+
```

Comment:
I hope soon means before deprecating the old settings? But probably not important as long as it is really fixed soon.

Reply:
I'm not sure, probably not. We are focusing on the job API and the next runner, and then I will work on this again. So we lowered the priority of this, for now. But yes, it will be really soon, I promise!